packages/taquito/src/contract/prepare.ts

```diff
@@ -30,7 +30,10 @@ export const createOriginationOperation = async ({
     );
   }
 
-  const contractCode = Array.isArray(code) ? code : ml2mic(code);
+  const order = ['parameter', 'storage', 'code'];
+  const contractCode = Array.isArray(code)
+    ? code.sort((a: any, b: any) => order.indexOf(a.prim) - order.indexOf(b.prim)) // Ensure correct ordering for RPC
+    : ml2mic(code);
```

Comment:
I guess my one concern here is that ml2mic produces the same order that is being defined and if not, `const schema = new Schema(contractCode[1].args[0])` could have an issue as well.

Reply:
This needs to be validated, perhaps something like this would cover it:
```
const contractCode = Array.isArray(code)
    ? code.sort((a: any, b: any) => order.indexOf(a.prim) - order.indexOf(b.prim))
    : ml2mic(code).sort((a: any, b: any) => order.indexOf(a.prim) - order.indexOf(b.prim));
```

Reply:
Hmm. From testing with `https://smartpy.io/demo/` they output the correct order when producing normal/naked Michelson, but when producing JSON, the order is incorrect.

I kinda feel like the `ml2mic` case is low risk. This issue probably originates from son encoders that don't preserve order.